Сухоруков Кирилл Евгеньевич

Так как на сайте "Кинопоиск" фильму нельзя выставить оценку 0, и пользователю доступны только значения от 1 до 10, значение рейтинга нельзя назвать непрерывными данными, но можно назвать порядковыми данными. Таким образом, для сравнения между собой двух рейтингов нельзя использовать Т-критерий Стьюдента, но можно использовать критерий Манна-Уитни

# Импорт библиотек и подключение Google Drive

In [94]:
import numpy as np
import pandas as pd
from scipy.stats import mannwhitneyu

In [95]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# Чтение файла и работа с данными

In [96]:
name_of_file= 'kinopoisk.csv'
path = '/content/drive/My Drive/Colab Notebooks/'+ name_of_file

In [97]:
data_ab = pd.read_csv(path, sep=',', decimal='.')

In [98]:
columns_mapping = {
    'film-item-rating-position__position':'Позиция в рейтинге',
    'selection-film-item-poster__rating':'Рейтинг 1',
    'selection-film-item-meta__name':'Название на русском',
    'rating__value':'Рейтинг 2',
    'rating__count':'Количество оценок',
    'selection-film-item-meta__meta-additional-item':'Страна',
    'selection-film-item-meta__meta-additional-item 2':'Жанр'
}

In [99]:
data_ab = data_ab.drop(['HaRNLxUnsJBznTziWnJLc', 'HaRNLxUnsJBznTziWnJLc href', 'selection-film-item-meta__link href', 'selection-film-item-meta__original-name'], axis='columns')

In [100]:
data_ab = data_ab.rename(columns=columns_mapping)

In [103]:
data_ab

Позиция в рейтинге  Рейтинг 1  ... Рейтинг 2 Количество оценок
0                     1        9.1  ...       8.9           700 424
1                     2        9.1  ...       8.9           790 435
2                     3        8.6  ...       8.8           488 267
3                     4        8.6  ...       8.8           475 722
4                     5        8.6  ...       8.8           526 518
..                  ...        ...  ...       ...               ...
245                 246        8.0  ...       8.0           225 690
246                 247        8.0  ...       8.0            71 092
247                 248        7.8  ...       8.0           349 746
248                 249        8.2  ...       8.0           128 533
249                 250        7.6  ...       8.0           192 787

[250 rows x 7 columns]

H0 - рейтинги не отличаются

In [101]:
result=mannwhitneyu(data_ab['Рейтинг 1'],data_ab['Рейтинг 2'])

In [102]:
print(f'P-value для нулевой гипотезы H0 имеет значение {result.pvalue}')

P-value для нулевой гипотезы H0 имеет значение 0.4512781866844041


Основываясь на значении P-value, можно сделать вывод о том, что при использовании критерия Манна-Уитни нулевая гипотеза H0 верна, так как P-value > 0.05. Значит, старый и новый рейтинги фильмов из "Топ 250 фильмов" сайта "Кинопоиск" не отличаются друг от друга.